contracts/LinearVesting.sol

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/**
+ * @title LinearVesting
+ * @dev
+ */
+contract LinearVesting is Ownable {
+    IERC20 token;
+    uint256 public start; // when the vesting starts
+    uint256 public cliff; // period in seconds
+    uint256 public duration; // vesting duration
+    address receiver; // recipient
+    uint256 amount; // number of tokens to be sent
+    uint256 totalAmount; // total amount of tokens locked in the contract
```

Comment:
Why are some variables public and others private ?
Is the token suppose to change ? You can also add constant or immutable.
We prefer to avoid leaving the "private by default".
Also there is no such thing as a private variable on most EVM, as soon as you verify the contract

Reply:
I've set all variables to public